src/pytest_plugins/consume/simulators/simulator_logic/test_via_sync.py

```diff
@@ -0,0 +1,540 @@
+                                ):
+                                    message = (
+                                        "Client returned unexpected validation error: "
+                                        f'got: "{payload_response.validation_error}" '
+                                        f'expected: "{payload.validation_error}"'
+                                    )
+                                    if strict_exception_matching:
+                                        raise LoggedError(message)
+                                    else:
+                                        logger.warning(message)
+
+                    except JSONRPCError as e:
+                        logger.info(f"JSONRPC error encountered: {e.code} - {e.message}")
+                        if payload.error_code is None:
+                            raise LoggedError(f"Unexpected error: {e.code} - {e.message}") from e
+                        if e.code != payload.error_code:
+                            raise LoggedError(
+                                f"Unexpected error code: {e.code}, expected: {payload.error_code}"
+                            ) from e
+
+                if payload.valid():
+                    with payload_timing.time(
+                        f"engine_forkchoiceUpdatedV{payload.forkchoice_updated_version}"
+                    ):
+                        # Send a forkchoice update to the engine
+                        version = payload.forkchoice_updated_version
+                        logger.info(f"Sending engine_forkchoiceUpdatedV{version}...")
+                        forkchoice_response = engine_rpc.forkchoice_updated(
+                            forkchoice_state=ForkchoiceState(
+                                head_block_hash=payload.params[0].block_hash,
+                            ),
+                            payload_attributes=None,
+                            version=payload.forkchoice_updated_version,
+                        )
+                        status = forkchoice_response.payload_status.status
+                        logger.info(f"Forkchoice update response: {status}")
+                        if forkchoice_response.payload_status.status != PayloadStatusEnum.VALID:
+                            raise LoggedError(
+                                f"unexpected status: want {PayloadStatusEnum.VALID},"
+                                f" got {forkchoice_response.payload_status.status}"
+                            )
+                        last_valid_block_hash = payload.params[0].block_hash
+
+        logger.info("All payloads processed successfully on client under test.")
+
+    # sync_payload creates the final block that the sync client will sync to
+    if not fixture.sync_payload:
+        pytest.fail("Sync tests require a syncPayload that is not present in this test.")
+
+    with timing_data.time("Send sync payload to client under test"):
```

Comment:
I think this could be reworded because "client under test" I thought at first that was the client that performs the sync.

Reply:
Hmm I've documented it throughout the file and logging as "client under test" (receiving the test payloads) and the "sync client". Let me know if you prefer something else here.